src/server/worker/master.go

```diff
@@ -255,6 +255,18 @@ func (a *APIServer) spoutSpawner(pachClient *client.APIClient) error {
 
 	var dir string
 
+	// Spouts typically have an open commit waiting for new data. So if the spout needs to be updated, and
+	// thus spoutSpawner is called, it might hang if the commit never gets closed. So to avoid this, we
+	// close open commits that we see here.
```

Comment:
Minor thing here, the comment is incorrect, we delete the commits below rather than closing them.